### PR TITLE
Adding testing package with OuiaSelectors

### PR DESCRIPTION
**Pull request by josejulio**
I'm sharing this utility that I found useful while testing. 

I didn't see any package (ideally, one package used as dev) where this could live, so I'm creating a new one. I didn't go all the way to configure it propertly, but it's something I could do if this is something that could live here.

This adds a selector utility to get Ouia DOM elements. I'm mixing it with the `react-testing-library` when I want to scope my queries to a subset of the DOM.

It can be used standalone (without the `@testing-library`) as follows:

```ts
expect(ouiaSelectors.getByOuia('PF4/Select')).toBeVisible();
```

with the `userEvent` (from the `@testing-library`)

```ts
userEvent.click(ouiaSelectors.getByOuia('PF4/Button', 'action'));
```

or mixing it up with other queries (from the `@testing-library`)

```ts
expect(
  getByLabelText(ouiaSelectors.getByOuia('Notifications/Integrations/Table'), /Enabled/i)
).toBeDisabled();
```

If this is something of interest, I could invest more time to configure it properly in this repository.

### CODE REVIEW

**Comment by josejulio** [comment]
@Hyperkid123 This one is ready for review:

-  Added README, documentation and tests
- Set version to 1.0.0 (unsure if should be using something else)
- Copied and adapted some common files across the projects (gitignore, npmignore, etc).

Unsure if there is a special setup for publishing that we must do.

**Comment by Hyperkid123** [comment]
> Unsure if there is a special setup for publishing that we must do.

@josejulio first publish has to be done manually. I publish a 0.0.1 before merging and we will let the bot to do a new major release.

File: packages/testing/package.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    "types": "index.d.ts",
+    "sideEffects": false,
+    "publishConfig": {
+        "access": "public"
+    },
+    "scripts": {
+        "build": "npm run build:js && npm run build:esm && npm run build:packages",
+        "build:packages": "node ../../scripts/build-packages.js --forceTypes",
+        "build:esm": "npx tsc --module es2015 --target es5",
+        "build:js": "npx tsc -p tsconfig.cjs.json"
```
Comment (Hyperkid123) [inline packages/testing/package.json:16]: The `npx` should not be used and is being fixed in another PR. Can you use TSC directly, please? For reference: 2c76dc06185f93bec73660e15338433b95eac6707317564c0f778f5b9abe446cR17

**Comment by Hyperkid123** [comment]
@josejulio I have also released a beta version of the package: https://www.npmjs.com/package/@redhat-cloud-services/frontend-components-testing release automation should now work after merging the PR.

**Comment by josejulio** [comment]
Addressed reviews, rebased and did a npm install to update whatever was missing.

**Review by Hyperkid123 — COMMENTED** [review]
I have just noticed a one little thing.

**Comment by josejulio** [comment]
@Hyperkid123 It's failing on something that seems unrelated: https://app.travis-ci.com/github/RedHatInsights/frontend-components/builds/246865156#L3708-L3726

**Comment by Hyperkid123** [comment]
Yeah that is unrelated. I've restarted the build.

**Review by Hyperkid123 — APPROVED** [review]

**Pull request merged**